Core/URLExtension.swift

```diff
@@ -20,8 +20,15 @@
 
 import Foundation
 
+
+
 extension URL {
     
+    enum URLProtocol: String {
+        case http
+        case https
+    }
+    
     private static let webUrlRegex = "^(https?:\\/\\/)?([\\da-z\\.-]+\\.[a-z\\.]{2,6}|(([\\d]+[.]){3}[\\d]+))([\\/]?[\\/:?=&#]{1}[\\%\\da-zA-Z_\\.-]+)*[\\/\\?]?$"
```

Comment:
Haha, yeah, this regex started off nice and simple enough and is slowly morphing into a beast. There's an outstanding asana task for a misclassified url. Whoever picks that up should kill this regex